src/include/btmem.h

```diff
@@ -420,11 +420,13 @@ struct __wt_page_modify {
 	 */
 	uint32_t write_gen;
 
-#define	WT_PAGE_LOCK(s, p)						\
-	__wt_spin_lock((s), &S2C(s)->page_lock[(p)->modify->page_lock])
-#define	WT_PAGE_UNLOCK(s, p)						\
-	__wt_spin_unlock((s), &S2C(s)->page_lock[(p)->modify->page_lock])
-	uint8_t page_lock;		/* Page's spinlock */
+#define	WT_PAGE_LOCK(s, p) 						\
+	__wt_spin_lock((s), &(p)->modify->page_lock)
+#define	WT_PAGE_TRYLOCK(s, p)						\
+	__wt_spin_trylock((s), &(p)->modify->page_lock)
+#define	WT_PAGE_UNLOCK(s, p) 						\
+	__wt_spin_unlock((s), &(p)->modify->page_lock)
+	WT_SPINLOCK page_lock;		/* Page's spinlock */
```

Comment:
We could save some bytes by using an unpadded version of `WT_SPINLOCK`.  Also note that this is only in modified pages.

I don't see any easy alternative: the earlier version where we had shared locks isn't going to work when reconciliation is holding pages locked and doing inserts into the lookaside table (and there are probably other cases I'm not aware of).